sisl/viz/plots/bands.py

```diff
@@ -366,38 +373,39 @@ def bands_wrapper(eigenstate, spin_index):
 
         # Get the eigenstates for all the available spin components
         bands_arrays = []
+        name = ["E"]
+        coords = [('band'), ]
+        coords_values = {"spin": spin_indices, "k": band_structure.lineark()}
+        if calc_spin_moments:
+            name.append("spin_moments")
+            coords.append(("band", "axis"))
+            coords_values["axis"] = ["x", "y", "z"]
+        for extra_var in extra_vars:
+            name.append(extra_var["name"])
+            coords.append(extra_var["coords"])
+
         for spin_index in spin_indices:
 
             # Non collinear routines don't accept the keyword argument "spin"
             spin_kwarg = {"spin": spin_index}
-            if self.spin.is_noncolinear:
+            if not self.spin.is_diagonal:
                 spin_kwarg = {}
-
-            spin_bands = band_structure.apply.dataarray.eigenstate(
-                wrap=partial(bands_wrapper, spin_index=spin_index),
-                **spin_kwarg,
-                coords=('band',),
-            )
+            
+            with band_structure.apply(pool=_do_parallel_calc, unzip=True) as parallel:
```

Comment:
you should probably be careful about this when the blas/lapack library is threaded. Then you'll saturate the machine quite fast?

Reply:
Is there some way of knowing whether the library is threaded?

Reply:
no, not really. But I think standard OS' are using a threaded library. In particular for conda installations this is true (MKL by default).